app/services/diagnostics/move_inspector.rb

```diff
@@ -216,9 +236,84 @@ def generate
         else
           @output << "(no person escort record events recorded)\n"
         end
+
+        @output << <<~ENDPER
+
+          YOUTH RISK ASSESSMENT
+          ---------------------
+        ENDPER
+
+        if move.profile&.youth_risk_assessment.present?
+          move.profile.youth_risk_assessment.tap do |yra|
+            @output << "id:\t#{yra.id}\n"
+            @output << "framework version:\t#{yra.framework&.version}\n"
+            @output << "framework_id:\t#{yra.framework_id}\n"
+            @output << "profile_id:\t#{yra.profile_id}\n"
+            @output << "move_id:\t#{yra.move_id}\n"
+            @output << "prefill_source_id:\t#{yra.prefill_source_id}\n"
+            @output << "section_progress:\n"
+            yra.section_progress.each do |section|
+              @output << "* #{section['key']}:\t#{section['status']}\n"
+            end
+            @output << "status:\t#{yra.status}\n"
+            @output << "created at:\t#{yra.created_at}\n"
+            @output << "updated at:\t#{yra.updated_at}\n"
+            @output << "confirmed at:\t#{yra.confirmed_at}\n"
+          end
+        else
+          @output << "(no youth risk assessment recorded)\n"
+        end
+      end
+
+      topics = [move]
+      topics << move.profile.person_escort_record if move&.profile&.person_escort_record.present?
+      topics << move.profile.youth_risk_assessment if move&.profile&.youth_risk_assessment.present?
+      notifications = Notification.where(topic: topics)
+
+      @output << <<~WEBHOOKS
+
+        WEBHOOK NOTIFICATIONS
+        ---------------------
+      WEBHOOKS
+      if notifications.webhooks.any?
+        @output << "#{'TYPE'.ljust(18)}\t#{'DELIVERED AT'.ljust(27)}\t#{'ATTEMPTS'.ljust(9)}\tENDPOINT\n"
+        notifications.webhooks.order(:created_at).each do |notification|
+          @output << "#{notification.event_type.ljust(18)}\t#{notification.delivered_at.to_s.ljust(27)}\t#{notification.delivery_attempts.to_s.ljust(9)}\t#{notification.subscription.callback_url}\n"
+        end
+      else
+        @output << "(no notifications recorded)\n"
+      end
+
+      @output << <<~EMAILS
+
+        EMAIL NOTIFICATIONS
+        -------------------
+      EMAILS
+      if notifications.emails.any?
+        @output << "#{'TYPE'.ljust(18)}\t#{'DELIVERED AT'.ljust(27)}\t#{'ATTEMPTS'.ljust(9)}\tEMAIL\n"
+        notifications.emails.order(:created_at).each do |notification|
+          @output << "#{notification.event_type.ljust(18)}\t#{notification.delivered_at.to_s.ljust(27)}\t#{notification.delivery_attempts.to_s.ljust(9)}\t#{notification.subscription.email_address}\n"
+        end
+      else
+        @output << "(no notifications recorded)\n"
       end
 
       @output
     end
+
+    def capture_events_errors(object)
+      event_valid = {}
+      object.generic_events.applied_order.each do |event|
+        event.trigger
```

Comment:
A bit concerned about this - events can have side effects applied when running them including updating other records. I get that we want to know that an event is "stuck" but we should be careful not to introduce any unexpected state changes purely as a result of running the diagnostics - i.e. we want to inspect not change the state. Perhaps wrap this in a reverted database transaction or something similar - or even better, add a "dry run" behaviour to events to ensure this is absolutely safe. It's also something that would be nice to see test coverage for as it's pretty critical behaviour.

Reply:
this was an excellent point well made; hopefully now resolved with the new `dry_run: true` parameter, and also by wrapping the runner in a transaction

Reply:
Nice work 👍